sdk/core/core/README.md

```diff
@@ -0,0 +1,123 @@
+# Azure SDK Core Library
```

Comment:
I would replace this section with information that's useful for an end user to understand the library.  Why do they care or need Azure.Core?  Here is what we can write:

"Azure.Core provides shared primitives, abstractions, and helpers for modern Azure SDK client libraries written in the C programming language.  These libraries follow the Azure SDK Design Guidelines for Embedded C.

Azure.Core allows client libraries to expose common functionality in a consistent fashion, so that once you learn how to use these APIs in one client library, you will know hot to use them in other client libraries.

The main shared concepts of Azure.Core (and so Azure SDK libraries using Azure.Core) include:

* Configuring service clients, e.g configuring retries, logging, tracing.
* Accessing HTTP response details.
* Calling long running operations (LROs).
* Paging and asynchronous streams.
* Reporting errors from service requests in a consistent fashion.
* Abstractions for representing Azure SDK credentials.

Below, you will find sections explaining these shared concepts in more detail."